zephyr/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+if(CONFIG_BOOTLOADER_MCUBOOT)
+  # Build a second bootloader image
+
+  set(MCUBOOT_BASE ${CMAKE_CURRENT_LIST_DIR}/..)
+
+  zephyr_add_executable(mcuboot require_build)
+
+  if (${require_build})
+    add_subdirectory(${MCUBOOT_BASE}/boot/zephyr ${CMAKE_CURRENT_BINARY_DIR}/mcuboot)
+
+    # TODO: Assert that the bootloader and image use the same key.
+
+    set(SIGNED_IMAGE signed.hex)
+
+    set_property(GLOBAL APPEND PROPERTY
+      extra_post_build_commands
+      COMMAND
+      ${PYTHON_EXECUTABLE}
+      ${MCUBOOT_BASE}/scripts/imgtool.py
+      sign
+      --key ${MCUBOOT_BASE}/${CONFIG_BOOT_SIGNATURE_KEY_FILE}
+      --header-size ${CONFIG_TEXT_SECTION_OFFSET}
+      --align       ${DT_FLASH_WRITE_BLOCK_SIZE}
+      --version 0.1       # TODO: Configurable?
```

Comment:
Do todo's around are supposed to be fulfilled in another PR?

Reply:
Yes, because there is need for sharing configurations between multiple images, this is not supported currently. 

This is one of two places where this functionality is needed for now, and both of them are in MCUBoot.